### PR TITLE
Preserve dict from original tests when creating codegen tests

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #103641

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78

### CODE REVIEW

File: test/inductor/test_cpp_wrapper.py [hunk 1]
```diff
@@ -112,6 +114,7 @@ def fn(self):
             tests.tearDownClass()
 
     fn.__name__ = test_name
+    fn.__dict__ = copy.deepcopy(func.__dict__)
```
Comment (albanD) [inline test/inductor/test_cpp_wrapper.py:117]: I'm a bit confused. Isn't `fn` a local function created just above? It's dict attr is not shared with anyone else and is fresh from creation?

Comment (ezyang) [reply test/inductor/test_cpp_wrapper.py:117]: The point is the original function may have had tags on it (e.g., expectedFailure is implemented this way), and when we make a new "copy" of the function that we created (as you say), we should propagate those tags.

Comment (albanD) [reply test/inductor/test_cpp_wrapper.py:117]: Ho sure, my bad, misread this...

**Review by albanD — APPROVED** [review]
SGTM!

**Pull request closed without merging**